### PR TITLE
Closes #2390 Store the query tool state

### LINKED ISSUE #2390: Query tool state
### Description
Evaluate the possibility to maintain the query tool state until a new layer is selected in TOC.
Currently, after performing a search, the query tool state is reset if the user close the feature grid, open the TOC or performs other operations.

### Please indicate if this issue is related to a bug or a new feature request
- [x] New Feature

### In case of New Feature

*Acceptance Criteria - AC*

- The state of the query panel should be maintained after a search until a new layer is selected in TOC

### PR DESCRIPTION
**Pull request by kappu72**
## Description
The featuregrid tool loses the advanced filter state on close. With this pull, the filterObj state is stored by layer in featuregrid state. The state is restored if present when the featuregrid is opened

## Issues
 - Fix #2390

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
On featuregrid close the filterObj is lost

**What is the new behavior?**
The filterObj is saved and restored from the state

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

### CODE REVIEW

File: web/client/actions/queryform.js [hunk 1]
```diff
@@ -255,9 +255,10 @@ function query(seachURL, data) {
     };*/
 }
 
-function reset() {
+function reset(skip) {
     return {
-        type: QUERY_FORM_RESET
+        type: QUERY_FORM_RESET,
+        skip
```
Comment (offtherailz) [inline web/client/actions/queryform.js]: I can't see skip anywhere else

Comment (kappu72) [reply web/client/actions/queryform.js]: It's used in query reducer

Comment (offtherailz) [reply web/client/actions/queryform.js]: please add a test that avoid any change on this. Explain well why it's necessary in the test.

File: web/client/reducers/queryform.js [hunk 2]
```diff
@@ -479,9 +480,12 @@ function queryform(state = initialState, action) {
...
         case LOAD_FILTER:
-            const {spatialField, filterFields, groupFields, crossLayerFilter} = (action.filter || initialState);
+            const {spatialField, filterFields, groupFields, crossLayerFilter, attributePanelExpanded, spatialPanelExpanded, crossLayerExpanded} = (action.filter || initialState);
             return {...state,
                     ...{
+                    attributePanelExpanded,
+                    spatialPanelExpanded,
+                    crossLayerExpanded,
                     spatialField: {
                         ...spatialField,
```
Comment (offtherailz) [inline web/client/reducers/queryform.js]: Thesting, seems that "attribute" is not restored correctly.
In fact the filter is created using "the_geom". Before this pull request attribute was filled by some triggered describeFeature (both for featuregrid or widgets) to set the correct spatial attribute. 

Make you sure that: 
 - spatialField geometry attribute name is correctly set for featuregrid, new widget and saved widget (during a second editing)
 - Also the crossLayerFilter has attribute value that is syncronized with the same logic. Make you sure that also this attribute is correctly set in any case of the ones listed above.  


**Add some unit tests to prove that all these things work**.

**Pull request merged**